sources/web/datalab/polymer/components/table-details/table-details.ts

```diff
@@ -0,0 +1,136 @@
+   */
+  public tableId: string;
+
+  private _table: gapi.client.bigquery.BigqueryTable | null;
+  private _busy = false;
+  private readonly DEFAULT_MODE = 'NULLABLE';
+
+  static get is() { return 'table-details'; }
+
+  static get properties() {
+    return {
+      _table: {
+        type: Object,
+        value: null,
+      },
+      creationTime: {
+        computed: '_computeCreationTime(_table)',
+        type: String,
+      },
+      lastModifiedTime: {
+        computed: '_computeLastModifiedTime(_table)',
+        type: String,
+      },
+      longTermTableSize: {
+        computed: '_computeLongTermTableSize(_table)',
+        type: String,
+      },
+      numRows: {
+        computed: '_computeNumRows(_table)',
+        type: String,
+      },
+      tableId: {
+        observer: '_tableIdChanged',
+        type: String,
+        value: '',
+      },
+      tableSize: {
+        computed: '_computeTableSize(_table)',
+        type: String,
+      },
+    };
+  }
+
+  _tableIdChanged() {
+    const matches = this.tableId.match(/^(.*):(.*)\.(.*)$/);
+    if (matches && matches.length === 4) { // The whole string is matched as first result
+      this._busy = true;
+      const projectId = matches[1];
+      const datasetId = matches[2];
+      const tableId = matches[3];
+
+      GapiManager.getTableDetails(projectId, datasetId, tableId)
+        .then((result) => {
+          this._table = JSON.parse(result.body) as gapi.client.bigquery.BigqueryTable;
+        }, (error) => console.log('Failed to get table details: ' + error.body))
+        .then(() => this._busy = false);
+    } else {
+      this._table = null;
+    }
+  }
+
+  _computeCreationTime(table: gapi.client.bigquery.BigqueryTable | null) {
+    if (table) {
+      return new Date(parseInt(table.creationTime, 10)).toLocaleString();
+    } else {
+      return '';
+    }
+  }
+
+  _computeLastModifiedTime(table: gapi.client.bigquery.BigqueryTable | null) {
+    if (table) {
+      return new Date(parseInt(table.lastModifiedTime, 10)).toLocaleString();
+    } else {
+      return '';
+    }
+  }
+
+  _computeNumRows(table: gapi.client.bigquery.BigqueryTable | null) {
+    return table ? parseInt(table.numRows, 10).toLocaleString() : '';
+  }
+
+  _computeLongTermTableSize(table: gapi.client.bigquery.BigqueryTable | null) {
+    return table ? this._bytesToReadableSize(table.numLongTermBytes) : '';
+  }
+
+  _computeTableSize(table: gapi.client.bigquery.BigqueryTable | null) {
+    return table ? this._bytesToReadableSize(table.numBytes) : '';
+  }
+
+  /**
+   * Converts the given number of bytes into a human readable string with units
+   * and a two-decimal-point number
+   */
+  _bytesToReadableSize(bytesStr: string) {
+    const kilo = 1024;
+    const units = ['B', 'KB', 'MB', 'GB', 'TB', 'PB', 'EB', 'ZB', 'YB'];
+    let bytes = parseInt(bytesStr, 10);
+    let level = 0;
+
+    while (bytes >= kilo) {
+      bytes /= kilo;
```

Comment:
Once you start dividing, this variable no longer contains the number of bytes. "significand" might be better, or even just "size" since that doesn't imply any units.